### ISSUE TITLE
Include exchange info and routing keys when declaring queues

### ISSUE THREAD

**Issue by epwalsh** [rusty-celery/rusty-celery#85]
Like https://docs.celeryproject.org/en/latest/userguide/routing.html#how-the-queues-are-defined. Only relevant to AMQP. See https://docs.rs/lapin/0.28.4/lapin/struct.Channel.html#method.exchange_declare

**Comment by alekhrycaiko** [comment]
hey @epwalsh, I was doing some thinking on this, as it looks like an interesting task. But, it'd be great to get some clarification around whether there's a simpler way to accomplish this (my current thoughts on solving this feel a bit far reaching). 

Since `task_routes` needs to support both `AMQP` and non `AMQP` cases, I believe `task_routes` would need to support an Option for an Exchange/Routing_key (since non AMQP cases don't need/require an exchange/routing key) on each Queue declaration.

Since we're supporting both cases, my thinking is that we could apply a pattern similar to how Celery [uses a kombu `Queue`](https://docs.celeryproject.org/en/latest/userguide/routing.html#manual-routing). 

e.g. Python from above link.
```
from kombu import Exchange, Queue

app.conf.task_queues = (
    Queue('feed_tasks',    routing_key='feed.#'),
    Queue('image_tasks',   exchange=Exchange('mediatasks', type='direct'), routing_key='image.compress'),
)
```

In Rust, this could maybe look like..e.g.
```
// Current implementation
task_routes = [ 
 '*' => 'celery'
]

// Becomes something similar to...
task_routes = [
 "*" => CeleryQueue::new( 
   name='celery',
   exchange_options=Option<insert_your_QueueDeclareOptions>,
   routing_key=Option<insert_your_string_for_routing_key>,
  ),
"add" => CeleryQueue::new(
   name = "add"
  // Maybe we default on `None` for `options` and `routing_key`
  )
}
```
That being said, this approach would effect `task_routes` pretty significantly, and would be a breaking change to the current API consumption paradigm. 

Any thoughts on this? Thanks for the patience/feedback 🙇

**Comment by epwalsh** [comment]
I think there's a way to do this that would keep backwards compatibility.

Like, if we implement `impl From<String> for CeleryQueue` or `impl From<&str> for CeleryQueue`, then I think the old way and new way of defining `task_routes` would both work.

**Comment by alekhrycaiko** [comment]
>  if we implement impl From<String> for CeleryQueue or impl From<&str> for CeleryQueue, then I think the old way and new way of defining task_routes would both work.

Good point :1st_place_medal: . I'll try to pull something together and see If I can get things to work/w the backwards compatibility.

 
Thanks for the 2nd set of eyes :+1:

**Comment by epwalsh** [comment]
Great! Looking forward to it 😄
